### ISSUE TITLE
AR Core - Augmented Images crashes on Samsung Galaxy S7 edge

### ISSUE THREAD

**Issue by SirLouLou** [google-ar/arcore-unreal-sdk#31]
Hi,

I downloaded the latest AR Core package (version 1.3), and build the Augmented Images application on my Samsung Galaxy S7 edge (SM-G935F). I did this all in Unreal Engine 4.20. 

After the build, the application launched and asked me permission for camera use. After I accepted, the screen goes black and crashed. And this happend multiple times, even used previous versions of package / engine but it didn't work.

I have AR Core installed and even have installed some AR Core applications. I even did a successful build on the 'AR Handheld' template from Unreal. So it has to do something with the package because I also tried to launch the HelloAR package and didn't work either.

I made a logcat for you:

[Log-augmentedimages.txt](https://github.com/google-ar/arcore-unreal-sdk/files/2216727/Log-augmentedimages.txt)

Tnx in advance!

**Comment by bopangzz** [comment]
It looks like it crashes because it has a corrupted AugmentedImageDatabase. Not sure how could that happened though. Can you attach the log when you build the application in Unreal editor?

**Comment by SirLouLou** [comment]
You're right, I see some errors in the build log. But don't know what those mean. Below you can find the build log:

[Log-BuildUE4.txt](https://github.com/google-ar/arcore-unreal-sdk/files/2221142/Log-BuildUE4.txt)

Thank you very much for the fast response!

**Comment by bopangzz** [comment]
It is possible that the arcoreimg tool couldn't add some of your images because it doesn't have enough feature point. Can you try run the arcoreimg tool in the command line and see if you have any images that has 0 score? 

See https://developers.google.com/ar/develop/c/augmented-images/arcoreimg for details.

**Comment by SirLouLou** [comment]
I tried, and that's not the problem. I used the standard images that came with the package. I didn't change anything from the package. I immediately build the package on my S7 edge and it continuously crashes after opening. The 'AR handheld template' works just fine, so i don't get it why this won't work :( 

Below some screenshots of my settings, if that may help:

![image](https://user-images.githubusercontent.com/25843481/43108127-924fec3c-8ee0-11e8-9654-11c299b72922.png)

![image](https://user-images.githubusercontent.com/25843481/43108182-d6a3acac-8ee0-11e8-83cd-0cfd162d3ef1.png)

![image](https://user-images.githubusercontent.com/25843481/43108196-e25d6c86-8ee0-11e8-9594-3ed5a6842cec.png)

![image](https://user-images.githubusercontent.com/25843481/43108203-ed889c34-8ee0-11e8-9adf-d1fbb59ddbbd.png)

**Comment by bopangzz** [comment]
Sorry I didn't realized you are building the Augmented Image sample. The image in that sample should work without problem. It looks like there is something wrong with the augmented image cooking process. The handheld ar works fine because it didn't load the augmented image database.

I wonder if it is because you have space in your Unreal Engine. Can you try put your Unreal Engine to a path without space and try package again? In particular, try remove space in the path in this command 

../../../Engine/Plugins/Runtime/GoogleARCore/Tools/Windows/arcoreimg.exe build-db --input_image_list_path=S:/Program Files/Unreal Engine/UE_4.20/Engine/Plugins/Runtime/GoogleARCore/Intermediate/ARCoreImgTemp/image_list.txt --output_db_path=S:/Program Files/Unreal Engine/UE_4.20/Engine/Plugins/Runtime/GoogleARCore/Intermediate/ARCore

**Comment by SirLouLou** [comment]
Hi bopangzz,

I want to let you know that I've read some simelar issues on this GIT and tried the following, but no results.


![image](https://user-images.githubusercontent.com/25843481/43160193-c9112854-8f84-11e8-8843-bd6acab32225.png)

![image](https://user-images.githubusercontent.com/25843481/43160199-cbe2e176-8f84-11e8-81d6-a18611bcf652.png)

I also duplicated the the original database and deleted the old one, like BryanChrisBrown did above (also changed the name to 'none'. But still crashes after launch on S7 edge:

![image](https://user-images.githubusercontent.com/25843481/43160223-da4db43e-8f84-11e8-86aa-847c670813ba.png)

![image](https://user-images.githubusercontent.com/25843481/43160230-e093a902-8f84-11e8-9f20-92d5173e4f5e.png)

------------------------------------------------------------

There is also an error that keeps repeating: 

Folder doesn't exist: 

![image](https://user-images.githubusercontent.com/25843481/43160267-fcd6a13c-8f84-11e8-8b06-665a50fd4a15.png)

And indeed, if I go to this path, the folder 'Tools' is missing in the GoogleARCore pluginsfolder. 

![image](https://user-images.githubusercontent.com/25843481/43160307-13785214-8f85-11e8-8b3e-6435951ba949.png)

Don't know why its missing, but every time when this error shoes up, the augmented image database is failing its build. So maybe there is a connection.

![image](https://user-images.githubusercontent.com/25843481/43160413-5b1dfd62-8f85-11e8-86c2-db5f7a73b13c.png)

For my master thesis I am working on a project to showcase the possibilities of Augmented Reality in Marketing and advertisement. In mid August, I will have my first presentation. I hope this issue will fix before the end of this week, so I can start building my project in UE4 :/

Nevertheless, I'm convinced that ARCore will be a breakthrough in 2019 and wish you developers all the luck with solving the problems that it will carry  with :) 🥇

**Comment by bopangzz** [comment]
Hmm, it looks like the binary build of Unreal Engine is missing the Tools folder inside GoogleARCore plugin. You might need to use the source code version of Unreal Engine and build your self. You can find the source code version of 4.20 here: https://github.com/EpicGames/UnrealEngine of if you want to get the latest GoogleARCore integration, you can use Google's version of Unreal Engine which you can find here: https://github.com/google-ar-unreal/UnrealEngine. Note that the Google's version is currently on 4.19 for now.

**Comment by bopangzz** [comment]
Also, if you want to keep use the binary version of Unreal Engine. You can use the source code version of Unreal Engine and run the setup.bat script inside it. That script should download the Tools folder in GoogleARCore plugin and you can try copy it to the binary version folder.

**Comment by SirLouLou** [comment]
Hi bopangzz,

Tnx for the tip! It's late at the moment. I will try this tomorrow morning and will update you on it.

**Comment by SirLouLou** [comment]
Hi bopangzz,

After a lot of hassle (deleting, installing and debugging the installation of the source code UE4 & Visual Studio) I finally got a working build of the augmented reality ARCore package. I even uploaded a custom image and everything is tracking very well on my Galaxy S7 Edge :)

I finally can go and build my AR application in Unreal.

Tnx for all the help! :)

**Comment by bopangzz** [comment]
@SirLouLou Glad you got it working! Just curious, are you using 4.20 source code version or Google's 4.19 branch?

**Comment by SirLouLou** [comment]
First I tried to install Google's 4.19 branch. But when trying to install the _GenerateProjectFiles.bat_ file, it gave me an error and didn't wanted to generate the projectfiles. I then installed the source code from Unreal version 4.20 but it gave me the same error. After re-installing, installing,... I did some research and found a solution on  the following post: [https://answers.unrealengine.com/questions/741965/cannot-generate-project-file-after-vs2017-update.html](url)

I needed to install the "NuGet package manager" in Visual Studio Installer, which Unreal never mentioned (or updated) in their documentation files. I tried another time and it did generate all the project files and after that I could successfully launch the UE4 source code and build the Augmented Images package on my phone.

In short: I sticked with the 4.20 source code version of Unreal, but I think that Google's branch would have worked fine, apart from the general error it gave on the  _GenerateProjectFiles.bat_ file, but that isn't an issue from your development team's side.

I hope this information may be useful to you, when solving other problems in this branch. If I could help in any way don't hesitate to message me :)

**Comment by bopangzz** [comment]
Yeah, good to know that 4.20 source code version works fine!

**Comment by bopangzz** [comment]
The following issues mentioned on this thread should be fixed in the ARCore 1.4 [release](https://github.com/google-ar/arcore-unreal-sdk/releases/tag/v1.4.0):
- AugmentedImageDatabase cooking process correctly handles engine and project paths that contain spaces.
- Fixed crash when starting ARCore session if AugmentedImageDatabase failed to cook correctly.
